core/src/main/java/io/questdb/cutlass/pgwire/PGConnectionContext.java

```diff
@@ -1403,89 +1560,87 @@ private void sendExecuteTail(int tail) throws PeerDisconnectedException, PeerIsS
+            case PG_INT4: // INT
+                bindVariableService.setInt(idx, Numbers.INT_NaN);
+                bindVariableSetters.add(intSetter);
+                bindVariableSetters.add(intTxtSetter);
                 break;
-            default:
+            case PG_INT8:
+                bindVariableService.setLong(idx, Numbers.LONG_NaN);
+                bindVariableSetters.add(longSetter);
+                bindVariableSetters.add(longTxtSetter);
+                break;
+            case PG_FLOAT4:
+                bindVariableService.setFloat(idx, Float.NaN);
+                bindVariableSetters.add(floatSetter);
+                bindVariableSetters.add(floatTxtSetter);
+                break;
+            case PG_INT2:
+                bindVariableService.setByte(idx, (byte) 0);
+                bindVariableSetters.add(byteSetter);
+                bindVariableSetters.add(byteTxtSetter);
+                break;
+            case PG_BOOL:
+                bindVariableService.setBoolean(idx, false);
+                bindVariableSetters.add(booleanSetter);
+                bindVariableSetters.add(booleanSetter);
                 break;
+            case PG_VARCHAR:
+                bindVariableService.setStr(idx, null);
+                bindVariableSetters.add(strSetter);
+                bindVariableSetters.add(strSetter);
+                break;
+            case PG_DATE:
+                bindVariableService.setDate(idx, Numbers.LONG_NaN);
+                bindVariableSetters.add(noopSetter);
+                bindVariableSetters.add(noopSetter);
+                break;
+            case PG_UNSPECIFIED:
+            case PG_TIMESTAMP:
+            case PG_TIMESTAMPZ:
+                // postgres JDBC driver does not seem to send
+                // microseconds with its text timestamp
+                // on top of this parameters such as setDate, setTimestamp
+                // cause driver to send UNSPECIFIED type
+                // QuestDB has to know types to resolve function linkage
+                // at compile time rather than at runtime.
+                bindVariableService.setDate(idx, Numbers.LONG_NaN);
+                bindVariableSetters.add(dateSetter);
+                bindVariableSetters.add(dateSetter);
+                break;
+            default:
+                throw SqlException.$(0, "unsupported parameter [type=").put(pgType).put(", index=").put(idx).put(']');
         }
-        send(PGConnectionContext.TAIL_NONE);
     }
 
     private void setupBindVariables(
             long lo,
             short pc,
-            @Transient ObjList<BindVariableSetter> bindVariableSetters
-    ) throws SqlException {
+            @Transient ObjList<BindVariableSetter> bindVariableSetters,
+            IntList bindVariableTypes) throws SqlException {
         bindVariableSetters.clear();
-        for (int j = 0; j < pc; j++) {
```

Comment:
`setupBindVariables` is called from `parse()`. It clears bind variable setters list and re-populates it with whatever bind variables are. The same `bindVariableSetters` list is also populated from `bind()` when portal name is specified. So it is possible that this list will have two sets of setters.

Obviously we need a unit test covering this scenario.